sonar-duplications/src/test/java/org/sonar/duplications/java/JavaTokenProducerTest.java

```diff
@@ -148,164 +150,175 @@ public void shouldNormalizeBinaryIntegerLiteral() {
+    assertThat(chunk("0xAFp-1")).containsExactly(NUMERIC_LITTERAL);
 
-    assertThat(chunk("0XAFp1f"), isNumericLiteral());
-    assertThat(chunk("0xAFp+1d"), isNumericLiteral());
-    assertThat(chunk("0XAFp-1D"), isNumericLiteral());
+    assertThat(chunk("0XAFp1f")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("0xAFp+1d")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("0XAFp-1D")).containsExactly(NUMERIC_LITTERAL);
   }
 
   /**
    * New in Java 7.
    */
   @Test
   public void shouldNormalizeNumericLiteralsWithUnderscores() {
-    assertThat(chunk("54_3L"), isNumericLiteral());
-    assertThat(chunk("07_7L"), isNumericLiteral());
-    assertThat(chunk("0b1_0L"), isNumericLiteral());
-    assertThat(chunk("0xF_FL"), isNumericLiteral());
-
-    assertThat(chunk("1_234."), isNumericLiteral());
-    assertThat(chunk("1_2.3_4"), isNumericLiteral());
-    assertThat(chunk(".1_234"), isNumericLiteral());
-    assertThat(chunk("1_234e1_0"), isNumericLiteral());
-
-    assertThat(chunk("0xA_F."), isNumericLiteral());
-    assertThat(chunk("0xA_F.B_C"), isNumericLiteral());
-    assertThat(chunk("0x1.ffff_ffff_ffff_fP1_023"), isNumericLiteral());
+    assertThat(chunk("54_3L")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("07_7L")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("0b1_0L")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("0xF_FL")).containsExactly(NUMERIC_LITTERAL);
+
+    assertThat(chunk("1_234.")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("1_2.3_4")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk(".1_234")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("1_234e1_0")).containsExactly(NUMERIC_LITTERAL);
+
+    assertThat(chunk("0xA_F.")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("0xA_F.B_C")).containsExactly(NUMERIC_LITTERAL);
+    assertThat(chunk("0x1.ffff_ffff_ffff_fP1_023")).containsExactly(NUMERIC_LITTERAL);
   }
 
   /**
    * <a href="http://java.sun.com/docs/books/jls/third_edition/html/lexical.html#3.10.3">Boolean Literals</a>
    */
   @Test
   public void shouldPreserveBooleanLiterals() {
-    assertThat(chunk("true false"), isTokens(new Token("true", 1, 0), new Token("false", 1, 5)));
+    assertThat(chunk("true false")).containsExactly(new Token("true", 1, 0), new Token("false", 1, 5));
   }
 
   /**
    * <a href="http://java.sun.com/docs/books/jls/third_edition/html/lexical.html#3.10.4">Character Literals</a>
    */
   @Test
   public void shouldNormalizeCharacterLiterals() {
-    assertThat("single character", chunk("'a'"), isStringLiteral());
-    assertThat("escaped LF", chunk("'\\n'"), isStringLiteral());
-    assertThat("escaped quote", chunk("'\\''"), isStringLiteral());
-    assertThat("octal escape", chunk("'\\177'"), isStringLiteral());
-    assertThat("unicode escape", chunk("'\\u03a9'"), isStringLiteral());
+    // single character
+    assertThat(chunk("'a'")).containsExactly(STRING_LITTERAL);
```

Comment:
Could have been translated into `.as("single character")` but not at all required, because if the test fails, we will have the line number anyways. So: 👍